src/rp2/plugin/report/ie/tax_report_ie.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2021 eprbell
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from datetime import date
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, List, Set, Tuple, cast
+
+from rp2.abstract_country import AbstractCountry
+from rp2.computed_data import ComputedData
+from rp2.entry_types import TransactionType
+from rp2.gain_loss import GainLoss
+from rp2.gain_loss_set import GainLossSet
+from rp2.logger import create_logger
+from rp2.plugin.report.abstract_ods_generator import AbstractODSGenerator
+from rp2.rp2_error import RP2TypeError
+
+LOGGER: logging.Logger = create_logger("tax_report_ie")
+
+
+class SheetNames(Enum):
+    AIRDROPS: str = "Airdrops"
+    CAPITAL_GAINS: str = "Capital Gains"
+    DONATIONS: str = "Donations"
+    GIFTS: str = "Gifts"
+    HARDFORKS: str = "Hard Forks"
+    INCOME: str = "Income"
+    INTEREST: str = "Interest"
+    INVESTMENT_EXPENSES: str = "Investment Expenses"
+    MINING: str = "Mining"
+    STAKING: str = "Staking"
+    WAGES: str = "Wages"
+
+
+_TEMPLATE_SHEETS_TO_KEEP: Set[str] = {f"__{item.value}" for item in SheetNames}
+
+_SHEET_TO_TYPES: Dict[str, Tuple[TransactionType, ...]] = {
+    SheetNames.AIRDROPS.value: (TransactionType.AIRDROP,),
+    SheetNames.CAPITAL_GAINS.value: (TransactionType.SELL,),
+    SheetNames.DONATIONS.value: (TransactionType.DONATE,),
+    SheetNames.GIFTS.value: (TransactionType.GIFT,),
+    SheetNames.HARDFORKS.value: (TransactionType.HARDFORK,),
+    SheetNames.INCOME.value: (TransactionType.INCOME,),
+    SheetNames.INTEREST.value: (TransactionType.INTEREST,),
+    SheetNames.INVESTMENT_EXPENSES.value: (
+        TransactionType.FEE,
+        TransactionType.MOVE,
+    ),
+    SheetNames.MINING.value: (TransactionType.MINING,),
+    SheetNames.STAKING.value: (TransactionType.STAKING,),
+    SheetNames.WAGES.value: (TransactionType.WAGES,),
+}
+
+_TYPE_TO_SHEET: Dict[TransactionType, str] = {
+    transaction_type: sheet_name for sheet_name, transaction_types in _SHEET_TO_TYPES.items() for transaction_type in transaction_types
+}
+
+
+class Generator(AbstractODSGenerator):
```

Comment:
This country-specific plugin is a great start and will probably be fairly easy to read for Irish tax preparers. However it's derived from the US-specific plugin, which uses [form 8949](https://www.irs.gov/pub/irs-pdf/f8949.pdf) format, US-specific form. As a future enhancement (not part of this PR), consider making the output of this plugin similar to what would be required by Irish tax regulations (essentially the Irish equivalent of form 8949).

Reply:
In Ireland there is no specific form required like form 8949. However, I thought it was useful form so produced an almost identical output (changed currency symbol and date format). Note that in the "Legend" it says "This report format is similar to that of the US IRS form 8949. Note it is NOT a form required by Revenue Ireland".

Reply:
Sounds good: actually form 8949 has a format that is understandable by most tax accountants, regardless of which country they work in.